Source/WebCore/dom/Document.cpp

```diff
@@ -2875,7 +2875,8 @@ void Document::willBeRemovedFromFrame()
     if (auto* pluginDocument = dynamicDowncast<PluginDocument>(*this))
         pluginDocument->detachFromPluginElement();
 
-    if (CheckedPtr page = this->page()) {
+    // Unable to ref the page as it may have started destruction.
+    if (WeakPtr page = this->page()) {
```

Comment:
Should this be `SingleThreadWeakPtr`?

Reply:
No need, with C++17, we don't need to specify template parameters. SingleThreadWeakPtr is only an alias to WeakPtr with specific template parameters.